File: 1704-determine-if-string-halves-are-alike/1704-determine-if-string-halves-are-alike.java
class Solution {
    public boolean isVowel(char ch) {
        if(ch == 'a' || ch == 'e' || ch == 'i' || ch == 'o' || ch == 'u' || ch == 'A' || ch
          == 'E' || ch == 'I' || ch == 'O' || ch == 'U') {
            return true;
        }
        return false;
    }
    public boolean halvesAreAlike(String s) {
        int cnt = 0;
        int i = 0, j = s.length()-1;
        while(i < j) {
            if(isVowel(s.charAt(i)))
                cnt += 1;
            if(isVowel(s.charAt(j))) 
                cnt -= 1;
            i += 1;
            j -= 1;
        }
        if(cnt == 0) 
            return true;
        return false;
    }
}

File: 1704-determine-if-string-halves-are-alike/1704-determine-if-string-halves-are-alike.py
class Solution:
    def isVowel(self,ch) -> bool:
        str = "aeiouAEIOU"
        return ch in str
        
    def halvesAreAlike(self, s: str) -> bool:
        i,j=0,len(s)-1
        cnt = 0
        while i < j:
            if self.isVowel(s[i]):
                cnt += 1
            if self.isVowel(s[j]):
                cnt -= 1
            i += 1
            j -= 1
        return (cnt == 0)